### PR TITLE
Plugins Admin: Avoid disabling auto-enabled apps

### LINKED ISSUE #97157: It's possible to disable the Explore Logs plugin, but it still shows Logs as an item under Explore in the sidebar. When clicking on it, unsurprisingly it says "Application Not Enabled." I'm sure this will confuse users.
> It's possible to disable the Explore Logs plugin, but it still shows Logs as an item under Explore in the sidebar. When clicking on it, unsurprisingly it says "Application Not Enabled." I'm sure this will confuse users.

from https://github.com/grafana/grafana/issues/96639

Steps to reproduce:

1. Go to any cloud stack
2. Go to https://svengrossmann.grafana.net/plugins/grafana-lokiexplore-app and disable the app
3. The `Explore -> Logs` section remains visible and errors when being clicked on

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Currently, the UI allows users to disable auto-enabled apps, that end up in a broken state. This PR prevents that:
 - The UI button to disable these apps is not rendered if the app is auto-enabled.
 - The HTTP and provision APIs now return an error when trying to disable an auto-enabled app.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/97157

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by academo — APPROVED** [review]

File: pkg/api/plugins.go [hunk 1]
```diff
@@ -258,6 +259,14 @@ func (hs *HTTPServer) UpdatePluginSetting(c *contextmodel.ReqContext) response.R
 		return response.Error(http.StatusNotFound, "Plugin not installed", nil)
 	}
 
+	p, found := hs.pluginStore.Plugin(c.Req.Context(), pluginID)
+	if !found {
+		return response.Error(http.StatusNotFound, "Plugin not found", nil)
+	}
```
Comment (xnyo) [inline pkg/api/plugins.go:265]: There's already a call to `hs.pluginStore.Plugin` right before this one, but the first returned argument is discarded. Can we re-use that one instead?

**Review by xnyo — COMMENTED** [review]
Great work! :raised_hands: Only one small comment

File: pkg/api/plugins.go (hunk 1 above)
Comment (andresmgot) [reply pkg/api/plugins.go:265]: good catch, I missed that one!

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**